src/odemis/driver/picoquant.py

```diff
@@ -660,6 +679,26 @@ def _acq_should_stop(self, timeout=None):
             pass
         return False
 
+    def _toggle_shutters(self, shutters, open):
+        """
+        Open/ close protection shutters.
+        shutters (list of string): the names of the shutters
+        open (boolean): True if shutters should open, False if they should close
+        """
+        axes = {}
+        fs = []
+        for sn in shutters:
+            logging.debug("Setting shutter %s to %s.", sn, open)
+            ax_name, closed_pos, open_pos = self._shutter_axes[sn]
+            shutter = self._shutters[sn]
+            if open:
+                axes[ax_name] = open_pos
+            else:
+                axes[ax_name] = closed_pos
+            fs.append(shutter.moveAbs(axes))
+
+        futures.wait(fs)
```

Comment:
I think it might actually be a good idea to fail if a shutter cannot move. Isn't it dangerous for the detector if we continue with an open shutter that should be closed?

Reply:
Well, at least on the current hardware, the fact that the picoquant reads the PMT output or not has no effect on the PMT status. If the shutter is opened, and there is too much light, the PMT will receive too much light, independently of whether the acquisition goes on or not. So it's not directly helping to avoid acquiring.

That said, I agree that it's important that the user knows that the shutter is left opened when it should be closed. The question is how is this information brought to the GUI, and shown to the user. Unfortunately, I don't think we have a good way now. Raising an exception would at best lead to raising an exception when stopping the dataflow, which would show up as an error in the GUI. However, that would also mean that the acquisition would be "half-stopped", and the user might have a hard time to interpret this unsubcribe failure to an issue with the TR shutter. 

I think that ideally, the failure to move should be signaled by the actuator, onto it's .state, and it'd be shown on the PC screen (either by the GUI, or by an independent program). For now, none of this takes place, but I thinks it's still best in picoquant to only log the error, and properly stop (or start) the acquisition.